src/Init/Data/Option/Basic.lean

```diff
@@ -15,7 +15,17 @@ public section
 
 namespace Option
 
-deriving instance DecidableEq for Option
+/- We write the instance manually instead of deriving it so that
+  it is coherent with `decidableEqNone` and `decidableNoneEq`. -/
```

Comment:
```suggestion
/- We write the instance manually so that it is coherent with `decidableEqNone` and
`decidableNoneEq`.

TODO: adjust the `deriving instance` handler to generate something coherent. -/
```